src/shells/shared/content-script.ts

```diff
@@ -1,43 +1,51 @@
+/**
+ * IIFE is here to isolate the variable scope of the content script
+ * as it might be injected into the same window/tab multiple times.
+ */
 (function() {
 	let port = chrome.runtime.connect({
 		name: "contentScript",
 	});
 
-	function onExtensionEvent(message: any) {
-		window.postMessage(
-			{
-				source: "preact-devtools-content-script",
-				...message,
-			},
-			"*",
-		);
-	}
-
+	/**
+	 * Keeps track whether the event piping done in the background script is ready
+	 * to accept events.
+	 */
 	let backendInitialized = false;
 
-	port.onMessage.addListener(onExtensionEvent);
+	/**
+	 * Setup message forwarding from port.onMessage to window.onMessage and vice-versa
+	 */
+	const cleanupPortMessagesForward = forwardPortMessagesToWindowPostMessage();
+	const cleanupWindowPostMessagesForward = forwardWindowPostMessagesToPortMessage();
+
+	/**
+	 * Cleanup message forwarding when the other end of the port disconnects
+	 */
 	port.onDisconnect.addListener(() => {
 		backendInitialized = false;
-		window.removeEventListener("message", onEvent);
-		port.onMessage.removeListener(onExtensionEvent);
+		cleanupWindowPostMessagesForward();
+		cleanupPortMessagesForward();
+		// TODO(Sven): Do we need to call `port.onDisconnect.removeListener` for this listenere here or
+		// is this taken care of since the port will be destroyed later anyways?
 	});
 
-	function onEvent(ev: MessageEvent) {
-		if (ev.source === window && ev.data) {
-			if (
-				ev.data.source === "preact-devtools-content-script" &&
-				ev.data.name === "initialized"
-			) {
-				backendInitialized = true;
-			} else if (backendInitialized && ev.data.source === "preact-devtools") {
-				port.postMessage({
-					name: ev.data.name,
-					data: ev.data,
-				});
+	/**
+	 * TODO(Sven): Why is this ping-ing needed? Found no handler of this event
+	 */
+	pingBackend();
```

Comment:
I poked around a bit and I can't find any justifications for that code. The extension works perfectly fine without this part and the if-block below... I think I copied that mostly from react: https://github.com/facebook/react/blob/master/packages/react-devtools-extensions/src/contentScript.js#L8